File: src/theme/tokens.ts
/**
 * Design tokens — the single source of truth for color, type, and spacing.
 * Values transcribed from docs/04_UIUX_SPEC.md §1. Hardcoding a color or a
 * spacing number anywhere else is a review-blocking rule.
 */

export type ColorScheme = 'light' | 'dark';

/**
 * `danger` is reserved exclusively for allergen warnings — not validation
 * errors, not destructive buttons. If red always means "this could hurt you,"
 * red keeps meaning it.
 */
export const palette = {
  light: {
    bg: '#FFFCF8',
    surface: '#FFFFFF',
    surfaceAlt: '#F5F0E8',
    text: '#1A1613',
    textMuted: '#6B6259',
    accent: '#D94F14',
    accentText: '#FFFFFF',
    ready: '#2E7D4F',
    near: '#C77D12',
    far: '#8A8079',
    danger: '#C62828',
    border: '#E5DDD2',
  },
  dark: {
    bg: '#151312',
    surface: '#221F1D',
    surfaceAlt: '#2C2825',
    text: '#F5F0E8',
    textMuted: '#A69C91',
    accent: '#FF7A3D',
    accentText: '#151312',
    ready: '#4CAF7D',
    near: '#E8A33D',
    far: '#9C938B',
    danger: '#FF6B6B',
    border: '#38332F',
  },
} as const;

export type ColorToken = keyof (typeof palette)['light'];

/**
 * The palette widened to `string`. `palette` is `as const` so that ColorToken
 * can be derived from it, but that also gives every value a literal type —
 * which makes the light and dark palettes mutually unassignable and breaks any
 * code that picks one at runtime.
 */
export type Palette = { readonly [K in ColorToken]: string };

/** System font stack only — a custom font costs bundle size and a layout shift. */
export const type = {
  display: { fontSize: 34, lineHeight: 40, fontWeight: '700' },
  title: { fontSize: 24, lineHeight: 30, fontWeight: '700' },
  heading: { fontSize: 19, lineHeight: 25, fontWeight: '600' },
  body: { fontSize: 17, lineHeight: 24, fontWeight: '400' },
  bodyStrong: { fontSize: 17, lineHeight: 24, fontWeight: '600' },
  caption: { fontSize: 14, lineHeight: 19, fontWeight: '400' },
  /** Cook mode only — readable from arm's length across a counter. */
  cookStep: { fontSize: 28, lineHeight: 38, fontWeight: '500' },
} as const;

/** 4pt base scale. */
export const space = { xs: 4, sm: 8, md: 16, lg: 24, xl: 32, xxl: 48 } as const;

export const radius = { sm: 8, md: 12, lg: 20, full: 999 } as const;

/**
 * Elevation (spec §1.3): `sm` for cards, `lg` for sheets.
 *
 * Dark mode is deliberately flat — a shadow on a near-black surface reads as
 * mud, so depth there comes from `border` instead. `elevation` is the Android
 * channel; the `shadow*` keys are iOS and web. Both are set because React
 * Native honours different ones per platform.
 */
export interface ElevationStyle {
  shadowColor: string;
  shadowOpacity: number;
  shadowRadius: number;
  shadowOffset: { width: number; height: number };
  /** The Android channel; the shadow* keys cover iOS and web. */
  elevation: number;
}

export interface ElevationSet {
  sm: ElevationStyle;
  lg: ElevationStyle;
}

export const shadow: Record<ColorScheme, ElevationSet> = {
  light: {
    sm: {
      shadowColor: '#1A1613',
      shadowOpacity: 0.06,
      shadowRadius: 8,
      shadowOffset: { width: 0, height: 2 },
      elevation: 2,
    },
    lg: {
      shadowColor: '#1A1613',
      shadowOpacity: 0.12,
      shadowRadius: 20,
      shadowOffset: { width: 0, height: 8 },
      elevation: 8,
    },
  },
  dark: {
    sm: {
      shadowColor: 'transparent',
      shadowOpacity: 0,
      shadowRadius: 0,
      shadowOffset: { width: 0, height: 0 },
      elevation: 0,
    },
    lg: {
      shadowColor: 'transparent',
      shadowOpacity: 0,
      shadowRadius: 0,
      shadowOffset: { width: 0, height: 0 },
      elevation: 0,
    },
  },
};

/**
 * The web build letterboxes the phone layout into a column of this width
 * rather than stretching it across a monitor. 430pt is the iPhone Pro Max
 * logical width — the largest phone the layout is designed for, so nothing
 * reflows differently in the browser than it does on device.
 */
export const layout = {
  mobileViewportMaxWidth: 430,
  tabletBreakpoint: 640,
  desktopBreakpoint: 960,
  desktopMaxWidth: 1180,
  desktopGutter: 32,
  desktopColumnGap: 24,
} as const;

/**
 * Minimum touch targets in pt. Cook mode is oversized deliberately — it is
 * operated with a knuckle or the back of a hand.
 */
export const touchTarget = { standard: 44, cookMode: 64, primaryCtaHeight: 56 } as const;

File: HomeChef/Theme/Tokens.swift
// Design tokens — the single source of truth for color, type, and spacing.
// Values transcribed from docs/04_UIUX_SPEC.md §1.
// Hardcoding a color or spacing number anywhere else is a review-blocking rule.

import SwiftUI

// MARK: - Color

/// `danger` is reserved exclusively for allergen warnings — not validation
/// errors, not destructive buttons. If red always means "this could hurt you,"
/// red keeps meaning it.
struct Palette {
    let bg:         Color
    let surface:    Color
    let surfaceAlt: Color
    let text:       Color
    let textMuted:  Color
    let accent:     Color
    let accentText: Color
    let ready:      Color
    let near:       Color
    let far:        Color
    let danger:     Color
    let border:     Color
}

extension Palette {
    static let light = Palette(
        bg:         Color(hex: "FFFCF8"),
        surface:    Color(hex: "FFFFFF"),
        surfaceAlt: Color(hex: "F5F0E8"),
        text:       Color(hex: "1A1613"),
        textMuted:  Color(hex: "6B6259"),
        accent:     Color(hex: "D94F14"),
        accentText: Color(hex: "FFFFFF"),
        ready:      Color(hex: "2E7D4F"),
        near:       Color(hex: "C77D12"),
        far:        Color(hex: "8A8079"),
        danger:     Color(hex: "C62828"),
        border:     Color(hex: "E5DDD2")
    )

    static let dark = Palette(
        bg:         Color(hex: "151312"),
        surface:    Color(hex: "221F1D"),
        surfaceAlt: Color(hex: "2C2825"),
        text:       Color(hex: "F5F0E8"),
        textMuted:  Color(hex: "A69C91"),
        accent:     Color(hex: "FF7A3D"),
        accentText: Color(hex: "151312"),
        ready:      Color(hex: "4CAF7D"),
        near:       Color(hex: "E8A33D"),
        far:        Color(hex: "9C938B"),
        danger:     Color(hex: "FF6B6B"),
        border:     Color(hex: "38332F")
    )
}

extension Color {
    init(hex: String) {
        var int: UInt64 = 0
        Scanner(string: hex.trimmingCharacters(in: .alphanumerics.inverted))
            .scanHexInt64(&int)
        let r = Double((int >> 16) & 0xFF) / 255
        let g = Double((int >>  8) & 0xFF) / 255
        let b = Double( int        & 0xFF) / 255
        self.init(.sRGB, red: r, green: g, blue: b, opacity: 1)
    }
}

// MARK: - Typography

/// System font stack only — a custom font costs bundle size and a layout shift.
enum TypeScale {
    static let display    = Font.system(size: 34, weight: .bold)
    static let title      = Font.system(size: 24, weight: .bold)
    static let heading    = Font.system(size: 19, weight: .semibold)
    static let body       = Font.system(size: 17, weight: .regular)
    static let bodyStrong = Font.system(size: 17, weight: .semibold)
    static let caption    = Font.system(size: 14, weight: .regular)
    /// Cook mode only — readable from arm's length across a counter.
    static let cookStep   = Font.system(size: 28, weight: .medium)
}

// MARK: - Spacing (4pt base scale)

enum Space {
    static let xs:  CGFloat = 4
    static let sm:  CGFloat = 8
    static let md:  CGFloat = 16
    static let lg:  CGFloat = 24
    static let xl:  CGFloat = 32
    static let xxl: CGFloat = 48
}

// MARK: - Corner radius

enum Radius {
    static let sm:   CGFloat = 8
    static let md:   CGFloat = 12
    static let lg:   CGFloat = 20
    static let full: CGFloat = 999
}

// MARK: - Touch targets

/// Minimum touch target sizes in points.
/// Cook mode is oversized deliberately — operated with a knuckle or the back of a hand.
enum TouchTarget {
    static let standard:        CGFloat = 44
    static let cookMode:        CGFloat = 64
    static let primaryCtaHeight: CGFloat = 56
}
